client/src/saves/deck.rs

```diff
@@ -18,9 +19,36 @@ impl Deck {
     }
 
     // todo: handle card classes, likely will need to take in game_io and namespace for resolving card classes
-    pub fn shuffle(&mut self, rng: &mut impl rand::Rng) {
+    pub fn shuffle(&mut self, game_io: &GameIO, rng: &mut impl rand::Rng, namespace: PackageNamespace) {
         use rand::seq::SliceRandom;
-
+        let globals = game_io.resource::<Globals>().unwrap();
         self.cards.shuffle(rng);
+        // If the deck is larger than 10 cards
+        if self.cards.len() < 10 {
```

Comment:
Just realized this may cause an infinite loop if not set to `<= 20`, there may also be an infinite loop if every card in the deck is a giga from permissive augments.

My understanding is we want to make sure there are no giga class cards within the first 10 cards, and we'll move any giga cards to a random position anywhere outside of that range. We'll need to make sure there are also free spots to put these cards in, and augments could allow every card to be giga.

Maybe we should change this function to track every index in range `10..len` that's not a giga in a vec, shuffle the vec, and pop out values for giga cards from `0..=9` to swap with, exiting if the vec is empty.